mdensemble/parsl.py

```diff
@@ -149,5 +149,77 @@ def config_factory(self, run_dir: PathLike) -> Config:
             app_cache=True,
         )
 
+class SunspotSettings(BaseComputeSettings):
+    """Configuration for running on Sunspot
 
-ComputeSettingsTypes = Union[LocalSettings, WorkstationSettings, PolarisSettings]
+    Each GPU tasks uses a single tile"""
+
+    name: Literal["sunspot"] = "sunspot"  # type: ignore[assignment]
+    label: str = 'htex'
+    worker_init: str = ""
+
+    num_nodes: int = 1
+    """Number of nodes to request"""
+    scheduler_options: str = ""
+    account: str
+    """The account to charge compute to."""
+    queue: str
+    """Which queue to submit jobs to, will usually be prod."""
+    walltime: str
+    """Maximum job time."""
+    retries: int = 0
+    """Number of retries upon failure."""
+    cpus_per_node: int = 208
+    strategy: str = "simple"
+
+    def config_factory(self, run_dir: PathLike) -> Config:
+        """Create a Parsl configuration for running on Sunspot."""
+        accel_ids = [
+            f"{gid}.{tid}"
+            for gid in range(6)
+            for tid in range(2)
+        ]
+        return Config(
+            executors=[
+                HighThroughputExecutor(
+                    label=self.label,
+                    available_accelerators=accel_ids,  # Ensures one worker per accelerator
+                    cpu_affinity="block",  # Assigns cpus in sequential order
+                    prefetch_capacity=0,
+                    max_workers=12,
+                    cores_per_worker=16,
+                    heartbeat_period=30,
+                    heartbeat_threshold=300,
+                    worker_debug=False,
+                    provider=PBSProProvider(
+                        launcher=MpiExecLauncher(
+                            bind_cmd="--cpu-bind",
+                            overrides="--depth=208 --ppn 1"
+                        ),  # Ensures 1 manger per node and allows it to divide work among all 208 threads
+                        #worker_init="""
```

Comment:
Are these no longer needed?

Reply:
So I wanted to keep these flags in the yaml file so they can be more modular. Similar to how it is for the polaris settings. I can remove these here